atom/browser/browser.h

```diff
@@ -158,6 +174,12 @@ class Browser : public WindowListObserver {
   // Add a custom task to jump list.
   bool SetUserTasks(const std::vector<UserTask>& tasks);
 
+  // Get the current Jump List settings.
+  v8::Local<v8::Value> GetJumpListSettings(mate::Arguments* args);
```

Comment:
The APIs under `atom::Browser` are also meant to be used by other parts of Electron code apart from V8 bindings, so methods under this class should not have dependency of `V8`. If an API is only exposed in JavaScript, it should be put in the `api::` classes directly. This is for the good of having a clean architecture.

So we should either make `Browser::GetJumpListSettings` and `Browser::SetJumpList` independent from V8, or just put them under `api::App`. And the `browser.cc` should not include any V8 code.

A few places are violating this rule and escaped review, and we will fix them.